compiler/rustc_codegen_llvm/src/asm.rs

```diff
@@ -706,6 +706,11 @@ fn llvm_fixup_input(
                 value
             }
         }
+        (InlineAsmRegClass::Mips(MipsInlineAsmRegClass::reg), Abi::Scalar(s)) => match s.value {
+            Primitive::Int(Integer::I8 | Integer::I16, _) => bx.sext(value, bx.cx.type_i32()),
+            // Primitive::Int(Integer::I8 | Integer::I16, false) => bx.zext(value, bx.cx.type_i32()),
```

Comment:
Should we do zero-extend for u8 or u16 ?

Reply:
No. The `asm!` documentation specifies that the upper bits of a register are undefined when you pass a smaller value in. So we can do either.

I just ran a few tests on godbolt and it turns out that zero-extension can always be done in 1 instruction while sign-extension sometimes needs 2 (unlike what my previous comment said). You should change this to always zero-extend.

Reply:
Hmm this is probably confusing since I change my mind halfway through the comment. But basically it doesn't really matter if you `sext` or `zext` from the spec point of view.